components/browser/search/src/main/java/mozilla/components/browser/search/SearchEngineManager.kt

```diff
@@ -66,11 +67,21 @@ class SearchEngineManager(
         val searchEngines = getSearchEngines(context)
 
         return when (name) {
-            EMPTY -> searchEngines[0]
+            EMPTY -> defaultSearchEngine ?: searchEngines[0]
             else -> searchEngines.find { it.name == name } ?: searchEngines[0]
         }
     }
 
+    /**
+     * Returns search engine corresponding to name passed to this method, or null if not found.
+     */
+    @Synchronized
+    fun getSearchEngineByName(context: Context, name: String): SearchEngine? {
+        val searchEngines = getSearchEngines(context)
+
+        return searchEngines.find { it.name == name }
+    }
```

Comment:
Will this be needed after we change the code to allow setting the name directly?

Reply:
I think this will still be needed in order for @sblatz to pass a non-default search engine to the invoke function?